### ISSUE TITLE
AWS: compilation failure regression

### ISSUE THREAD

**Issue by kubukoz** [disneystreaming/smithy4s#293]
If you try to compile a service like Kinesis, you now get compilation errors due to a naming clash (Timestamp: aws vs smithy4s):

```
[error] /Users/kubukoz/projects/demos/target/scala-2.13/src_managed/main/com/amazonaws/kinesis/Consumer.scala:17:72: type mismatch;
[error]  found   : com.amazonaws.kinesis.Timestamp
[error]     (which expands to)  com.amazonaws.kinesis.Timestamp.Type
[error]  required: smithy4s.Timestamp
[error]     Timestamp.schema.required[Consumer]("ConsumerCreationTimestamp", _.consumerCreationTimestamp).addHints(smithy.api.Required(), smithy.api.Documentation("<p></p>")),
[error]                                        
```

I think it first appeared in 0.13.1. If you go back to 0.12.16 the codegen fails:

```
[error] stack trace is suppressed; run last Compile / smithy4sCodegen for the full output
[error] (Compile / smithy4sCodegen) software.amazon.smithy.model.validation.ValidatedResultException: Result contained ERROR severity validation events: 
[error] [ERROR] com.amazonaws.kinesis#DescribeStream: Unable to resolve trait `smithy.waiters#waitable`. If this is a custom trait, then it must be defined before it can be used in a model. | Model /Users/kubukoz/projects/demos/src/main/smithy/kinesis.json:508:36
[error] [ERROR] com.amazonaws.kinesis#Kinesis_20131202: Unable to resolve trait `aws.api#service`. If this is a custom trait, then it must be defined before it can be used in a model. | Model /Users/kubukoz/projects/demos/src/main/smithy/kinesis.json:1270:28
[error] [ERROR] com.amazonaws.kinesis#Kinesis_20131202: Unable to resolve trait `aws.auth#sigv4`. If this is a custom trait, then it must be defined before it can be used in a model. | Model /Users/kubukoz/projects/demos/src/main/smithy/kinesis.json:1277:27
[error] [ERROR] com.amazonaws.kinesis#Kinesis_20131202: Unable to resolve trait `aws.protocols#awsJson1_1`. If this is a custom trait, then it must be defined before it can be used in a model. | Model /Users/kubukoz/projects/demos/src/main/smithy/kinesis.json:1280:37
```

and 0.12.15 works fine.

Build:

```scala

import smithy4s.codegen.Smithy4sCodegenPlugin

val root = project
  .in(file("."))
  .enablePlugins(Smithy4sCodegenPlugin)
  .settings(
    scalaVersion := "2.13.7",
    libraryDependencies ++= Seq(
      "com.disneystreaming.smithy4s" %% "smithy4s-aws" % smithy4sVersion.value
    )
  )
```

Reproduction: https://github.com/kubukoz/demos/tree/smithy4s-aws-regression

**Comment by yisraelU** [comment]
resolved via #366

**Comment by kubukoz** [comment]
Nice!
